include/mgmt/mcumgr/buf.h

```diff
@@ -8,18 +8,18 @@
 #define ZEPHYR_INCLUDE_MGMT_BUF_H_
 
 #include <inttypes.h>
-#include "tinycbor/cbor.h"
-#include "tinycbor/cbor_buf_writer.h"
+#include <zcbor_common.h>
+#include <zcbor_encode.h>
 struct net_buf;
 
 struct cbor_nb_reader {
-	struct cbor_decoder_reader r;
 	struct net_buf *nb;
+	zcbor_state_t zs[4];
```

Comment:
Significance of 4?

Reply:
Number of zcbor state backups, will add Kconfig for this.

Reply:
Added Kconfigs for that